tutorial3_distributions/demo/main.js

```diff
@@ -71,63 +66,66 @@ function init() {
-    .attr("transform", `translate(0,${height - margin.bottom})`)
+    .attr("transform", `translate(0,${constants.height - constants.margin.bottom})`)
     .call(xAxis);
 
   // add the xAxis
   svg
     .append("g")
     .attr("class", "y-axis")
-    .attr("transform", `translate(${margin.left},0)`)
+    .attr("transform", `translate(${constants.margin.left},0)`)
     .call(yAxis);
 
   draw();
 }
 
+// DRAW FUNCTION
 function draw() {
   // we call this everytime there is an update to the data/state
 
-  // filter the data for the selectedState
-  let filteredData = data;
-  // if there is a selectedState, filter the data before drawing it
-  if (selectedState) {
-    filteredData = data.filter(d => d.state === selectedState);
+  // filter the data for the selectedParty
+  let filteredData = state.data;
+  // if there is a selectedParty, filter the data before drawing it
+  if (state.selectedParty) {
+    filteredData = state.data.filter(d => 
+      state.selectedParty === "Both" 
+    || d.party === state.selectedParty);
+    console.log("new filtered data includes", filteredData.length, "politicians");
   }
 
-  /** create a `g`, or group element
-   * This `g` will contain within it the `circle` and `text` elements
-   * We do it this way so that we don't have to position the `circle` and `text` separtely
-   * (since we want to keep the label tied to the circle it is annotating)
-   */
-  //
-  const dotgroup = svg
+  const dot = svg
     .selectAll(".dot")
-    .data(filteredData) // map our data to all the `.dot` elements
-    .join("g")
-    .attr("class", "dot")
-    .attr(
-      // move the group to be in the right position
-      "transform",
-      d =>
-        // translate(x-position, y-position)
-        `translate(${xScale(d.ideology_rating)},
-        ${yScale(d.environmental_rating)})`
-    );
-
-  // add the circles into each group
-  dotgroup.append("circle").attr("r", radius);
-
-  // add text into each group
-  dotgroup
-    .append("text")
-    .attr("class", "hover-label")
-    .attr("dy", -radius) // offset the text a bit so it doesn't overlab with circle
-    .text(d => d.name);
+    .data(filteredData, d => d.name)
+    .join(
+      enter => enter.append("circle")
+        .attr("class", "dot")
+        .attr("cx", d => xScale(d.ideology_rating))
+        .attr("cy", d => yScale(d.environmental_rating))
+        .attr("r", constants.radius)
+        .attr("stroke", "lightgrey")
+        .attr("fill", "green")
+        .call(enter => enter.transition()
+          .transition()
+          .duration(500)
+          .attr("fill", "blue")),
+      update => update
+        .attr("fill", "orange")
+        .call(update => update.transition()
+          .transition()
+          .duration(500)
+          .attr("fill", "blue")),
+      exit => exit
+        .attr("fill", "red")
+        .call(exit => exit.transition()
+          .duration(100)
+          .delay(d => 500 * d.ideology_rating)
+          .remove())
```

Comment:
this shows that you can access data within the `duration` or `delay`, and will "wipe" from left to right. It also includes a change to red before it gets removed.